pkg/config/config.go

```diff
@@ -81,14 +86,77 @@ func (c *Config) ConfigureRules() {
 }
 
 func loadConfig(filename string) (c Config, err error) {
+	if isValidURL(filename) {
+		// if fileName is a valid URL, we will download and set it to the config
+		log.Debug().Str("url", filename).Msg("Downloading file from")
+		// hardcoding this file and saving to root directory
+		downloadedFile := "downloadedRules.yaml"
+		err := DownloadFile(downloadedFile, filename)
+		if err != nil {
+			return c, err
+		}
+		filename = downloadedFile
+		log.Debug().Str("filename", filename).Msg("Saved remote config to local file.")
+	}
+
+	// TO DO = Need to add more error handling on reading/validating yamlFile file - issue # 16
 	yamlFile, err := ioutil.ReadFile(filename)
+	log.Debug().Str("filename", filename).Msg("Adding custom ruleset from")
 	if err != nil {
 		return c, err
 	}
-
 	return c, yaml.Unmarshal(yamlFile, &c)
 }
 
+// isValidUrl tests a string to determine if it is a valid URL or not
+func isValidURL(toTest string) bool {
+	_, err := url.ParseRequestURI(toTest)
+	if err != nil {
+		return false
+	}
+
+	u, err := url.Parse(toTest)
+	if err != nil || u.Scheme == "" || u.Host == "" {
+		return false
+	}
+	log.Debug().Str("remoteConfig", toTest).Msg("Valid URL for remote config.")
+	return true
+}
+
+// downloads file from url to set filepath
+func DownloadFile(filepath string, url string) error {
+	// Removing gosec lint warning - as we have to pass in user specified url for remote config
+	var client = &http.Client{
+		Timeout: time.Second * 10,
+	}
+	ctx := context.Background()
+	req, _ := http.NewRequestWithContext(ctx, http.MethodGet, url, nil)
+	resp, err := client.Do(req)
+	if err != nil {
+		return err
+	}
+
+	// only parse response body if it is in the response is in the 2xx range
+	if resp.StatusCode >= 200 && resp.StatusCode <= 299 {
+		log.Debug().Int("HTTP Response Status:", resp.StatusCode).Msg("Valid URL Response")
+		defer resp.Body.Close()
+
+		// Create the file
+		out, err := os.Create(filepath)
+		if err != nil {
+			return err
+		}
+		defer out.Close()
+
+		log.Debug().Int("here", resp.StatusCode).Msg("help")
+		// Write the body to file
+		_, err = io.Copy(out, resp.Body)
+		return err
+	} else {
+		return fmt.Errorf("unable to download remote config from url - http response code: %v", resp.StatusCode)
```

Comment:
Totally agree with returning the error here -- maybe we should also log.Error() the msg/body of the http response here as well? Just thinking it might provide more detailed information in case the remote server sent relevant info in the response body

Reply:
Added in the response body to the error message!